### PR TITLE
feat: add example for using javascript SDK

### LINKED ISSUE #816: Create instructions on how to create expose ports necessary to create an SDK connection to both the Consenus Nodes plus the Mirror Node
```[tasklist]
### Tasks
- [ ] Create a `docs/content/User/SDK.md`
- [ ] Add a 'SDK Usage' + ref: "/User/SDK.md" element in `docs/data/menu/main.yml`
- [ ] Reference instructions on [how to create a client node](https://docs.hedera.com/hedera/tutorials/consensus/submit-your-first-message) and fill out the consensus node network manually, similar to how we do in [solo using Client.fromConfig()](https://github.com/hashgraph/solo/blob/537398caf6cdcbce762ad6b6e5ae28a47dff860b/src/core/account_manager.ts#L240-L240)
- [ ] it looks like the mirror node can also be configured inside there: https://github.com/hashgraph/hedera-sdk-js/blob/ac4914112cb80deadd12ac350e19e27b06d43bf1/src/client/Client.js#L54-L54, or alternatively use nodeClient.setMirrorNetwork()
- [ ] we should add a test case to verify that it works correctly
- [ ] we need to ensure that we are providing the steps to expose the port forwards for both the consenus nodes as well as the mirror node network
```

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

Changes to Taskfile.yaml
- Fix the issue of node name starting from node0 instead of node1
- Remove installation task per user suggest, since user may already installed own tool chains even not enabled in search path, should not force to install again
- Add relay deploy and destroy task
- Add port-forward tasks

Added a unit test to instantiate SDK client to interact with solo

Added SDK.md about how to use Hedera Javascript SDK with solo

https://github.com/hashgraph/solo/blob/00816-D-client-doc-example/docs/content/User/SDK.md

By using environment variable OPERATOR_ID and OPERATOR_KEY, end user can start
running SDK examples immediately after deploying solo.

### Related Issues

* Closes #816

### CODE REVIEW

**Review by mishomihov00 — APPROVED** [review]
Reviewed only these files:
`.github/workflows/flow-task-test.yaml`
`.github/workflows/zxc-e2e-test.yaml`

**Comment by jeromy-cannon** [comment]
Is this ready for review?  you have it in open status and not draft status.  You also have a lot of merge conflicts.

**Comment by JeffreyDallas** [comment]
> Is this ready for review? you have it in open status and not draft status. You also have a lot of merge conflicts.

ready for review

File: .github/workflows/script/solo_smoke_test.sh [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+{
+  echo "Create test account with solo network"
+  cd solo
+
+  # create new account and extract account id
+  npm run solo-test -- account create -n solo-e2e --hbar-amount 100 > test.log
+  export OPERATOR_ID=$(grep "accountId" test.log | awk '{print $2}' | sed 's/"//g'| sed 's/,//g')
+
+  # get private key of the account
+  npm run solo-test -- account get -n solo-e2e --account-id ${OPERATOR_ID} --private-key > test.log
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:203]:
```suggestion
  npm run solo-test -- account get -n solo-e2e --account-id ${OPERATOR_ID} --private-key >> test.log
```

File: .github/workflows/script/solo_smoke_test.sh [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+function enable_port_forward ()
+{
+  kubectl port-forward -n solo-e2e svc/haproxy-node1-svc 50211:50211 > /dev/null 2>&1 &
+  kubectl port-forward -n solo-e2e svc/hedera-explorer 8080:80 > /dev/null 2>&1 &
+  kubectl port-forward -n solo-e2e svc/relay-node1-hedera-json-rpc-relay 7546:7546 > /dev/null 2>&1 &
+}
+
+# calling local node generate-accounts function and extract private key
+# from the output, after some manipulation, add the private key to hardhat.config.js
+function create_account_and_extract_key ()
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:23]: remove

File: .github/workflows/script/solo_smoke_test.sh [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    echo "File private_key_without_quote_final.txt exists and not empty"
+    return 0
+  else
+    echo "File private_key_without_quote_final.txt does not exist or empty"
+    return 1
+  fi
+}
+
+function clone_smart_contract_repo ()
+{
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:65]: remove

File: .github/workflows/script/solo_smoke_test.sh [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  if [ -d "hedera-smart-contracts" ]; then
+    echo "Directory hedera-smart-contracts exists."
+  else
+    echo "Directory hedera-smart-contracts does not exist."
+    git clone https://github.com/hashgraph/hedera-smart-contracts --branch only-erc20-tests
+  fi
+}
+
+function clone_sdk_repo ()
+{
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:76]: remove

File: .github/workflows/script/solo_smoke_test.sh [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    cd hedera-sdk-js
+    npm install --save @hashgraph/sdk
+    cd -
+  fi
+}
+
+# clone hedera local node repo and call function create_account_and_extract_key
+# to extract private keys
+function clone_local_node_repo ()
+{
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:92]: remove

File: .github/workflows/script/solo_smoke_test.sh [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+  echo "Function $function_name failed after 5 retries"
+  exit 1
+}
+
+function setup_smart_contract_test ()
+{
+  echo "Setup smart contract test"
+  CONTRACT_TEST_KEYS=$(cat hedera-local-node/private_key_without_quote_final.txt)
+
+  cd hedera-smart-contracts
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:133]: just get a copy of the smart contracts we want and check them in our repo in a good location

File: .github/workflows/script/solo_smoke_test.sh [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    ps -ef |grep port-forward
+  done &
+}
+
+function start_background_transactions ()
+{
+  echo "Start background transaction"
+  # generate accounts every 3 seconds as background traffic for two minutes
+  # so record stream files can be kept pushing to mirror node
+  cd hedera-local-node
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:165]: just use Solo to generate the accounts, we have a `solo account create` function.  can just call it multiple times.   Or, even better would be to enable the mirror node monitor, my understanding is that it wil automatically push transactions.

File: .github/workflows/script/solo_smoke_test.sh [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+  cd -
+}
+
+function start_contract_test ()
+{
+  cd hedera-smart-contracts
+  echo "Wait a few seconds for background transactions to start"
+  sleep 5
+  echo "Run smart contract test"
+  npm run hh:test
```
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:180]: not sure what this is doing, but maybe just pull into our repo somehow.

File: tsconfig.json [hunk 9]
```diff
@@ -14,7 +14,8 @@
     "target": "ES2022",                                  /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
     "lib": [
       "ES2022",
-      "dom"
+      "dom",
+      "webworker"
```
Comment (jeromy-cannon) [inline tsconfig.json:18]: what is this for?

File: .github/workflows/script/solo_smoke_test.sh
Comment (jeromy-cannon) [inline .github/workflows/script/solo_smoke_test.sh:1]: Nathan and I had a conversation and are concerned about the support for this.  Maybe split this out, and put this into a separate PR so that the rest of this PR can merge.

**Comment by JeffreyDallas** [comment]
I remove the smoke test from workflow

File: .github/workflows/script/relay_smoke_test.sh
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply .github/workflows/script/relay_smoke_test.sh:1]: no problem to delete it now, I have a different branch to keep a copy

**Review by jeromy-cannon — APPROVED** [review]
